### PR TITLE
Decorate class

### PR DESCRIPTION
**Pull request by fredo509**
For this millestone I : 

- Create a class Nameable  with a method `correct_name` that does the following:
- Add a method` validate_name` that uses the Corrector instance to validate and save the corrected name.

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name(name)
+    name[0...10].capitalize
+  end
```
Comment (Kamba56) [inline nameable.rb]: - [ ] According to the project requirements, kindly make the correct_name method in the Nameable class raise a `NotimplementedError`. The method does not need to receive any argument, it should only raise the error.

- [ ] As demanded by the project requirements, kindly create a Decorator class that inherits the Nameable class. In the constructor assign a nameable object from params to an instance variable. Implement the correct_name method that returns the result of the correct_name method of the @nameable. The correct name method in the Decorator class should look like
```ruby
def correct_name
      @nameable.correct_name
    end
```

- [ ] Kindly add a CapitalizeDecorator class that inherits the Decorator class as demanded by project requirements. You should also implement a method correct_name that capitalizes the output of @nameable.correct_name.

- [ ] Kindly add a TrimmerDecorator class that inherits the Decorator class as demanded by project requirements. Also, implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

File: person.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
+require_relative('./nameable')
+
 class Person
```
Comment (Kamba56) [inline person.rb]: - [ ] As demanded by project requirements, kindly make sure the Person class inherits the Nameable class.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @fredo509, 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great job adding the nameable class👏👏👏
- [x] All linters passed

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zainsadaqat — APPROVED** [review]
# Pull Request Approved ✅ 🎊 🎉 
## Hello @fredo509  👋 

### Things to Appreciate
- [ ] Worked on features requested by the previous code reviewer

### Your project is complete!  There is nothing else to say other than it's time to merge it.

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 ☑️ 



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**